Source/BugsnagClient.m

```diff
@@ -206,9 +207,14 @@ @interface BugsnagClient ()
 @property(nonatomic, strong) BugsnagErrorReportApiClient *errorReportApiClient;
 @property(nonatomic, strong, readwrite) BugsnagSessionTracker *sessionTracker;
 @property (nonatomic, strong) BSGOutOfMemoryWatchdog *oomWatchdog;
+@property (nonatomic, strong) BugsnagPluginClient *pluginClient;
 @property (nonatomic) BOOL appCrashedLastLaunch;
 @end
 
+@interface BugsnagConfiguration ()
+@property(nonatomic, readwrite, strong) NSMutableSet *plugins;
+@end
+
```

Comment:
Is this the correct place for the declaration? Why not in `BugsnagConfiguration.h`?

Reply:
This is a [class extension](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/CustomizingExistingClasses/CustomizingExistingClasses.html#//apple_ref/doc/uid/TP40011210-CH6-SW3) which is used in Objective-C to hide implementation details. `BugsnagClient` will eventually need to be a public class and when it is, this property should be hidden.

Reply:
Put another way: this declaration gives `BugsnagClient` visibility of `BugsnagConfiguration`'s `plugins` property.